### DISCUSSION TITLE
Getting "It seems there is already an App at ..." even when trying to uninstall

### DISCUSSION THREAD

**Discussion by Apromixately** [Homebrew/discussions#4758] [Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.7
ORIGIN: https://github.com/Homebrew/brew
HEAD: d4444b563e24ac7c05a93121c464c02dfa04d44f
Last commit: 5 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: d57a3e8393e451bf4a049bb11bd1437c360ee4be
Core tap last commit: 2 days ago
Core tap branch: master
Core tap JSON: 01 Sep 10:04 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.42.0 => /opt/homebrew/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 13.5.1-arm64
CLT: 14.3.1.0.1.1683849156
Xcode: 14.3.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  python-certifi
  six

Warning: You have an unnecessary local Cask tap.
This can cause problems installing up-to-date casks.
Please remove it by running:
  brew untap homebrew/cask

Warning: You have an unnecessary local Core tap!
This can cause problems installing up-to-date formulae.
Please remove it by running:
 brew untap homebrew/core
```

### Description of issue

Trying to run pgadmin4 it says it is incomplete or damaged. Trying to upgrade or even uninstall gives: `Error: pgadmin4: It seems there is already an App at '/opt/homebrew/Caskroom/pgadmin4/7.4/pgAdmin 4.app'.`

What is going on or what am I supposed to do here?

### Relevant casks

pgadmin4

**Comment by Makki93** [comment]
I have the same issue. Can you open up an official issue? Not only a discussion?
Not only for pgadmin, but other apps, like pomatez.

**Comment by tony19** [comment, marked as answer]
I ran into the same problem (running Homebrew 4.1.19) after aborting a `brew --cask` installation, which left that app "half-installed".

Trying to uninstall via `brew` resulted in the error you've mentioned. The trick that worked for me was to use the `--force` flag:

```
brew remove --force --cask <CASK_NAME>
```

That changed the error into a warning and proceeded with the clean uninstall.

BEFORE:
```
$ brew remove phoenix
==> Uninstalling Cask phoenix
==> Removing login item /Applications/Phoenix.app
==> Removing login item Phoenix
Error: It seems there is already an App at '/opt/homebrew/Caskroom/phoenix/4.0.0/Phoenix.app'.
```

AFTER:
```
                 👇
$ brew remove --force phoenix
==> Uninstalling Cask phoenix
==> Removing login item /Applications/Phoenix.app
==> Removing login item Phoenix
Warning: It seems there is already an App at '/opt/homebrew/Caskroom/phoenix/4.0.0/Phoenix.app'; overwriting.
==> Removing App '/opt/homebrew/Caskroom/phoenix/4.0.0/Phoenix.ap
==> Purging files for version 4.0.0 of Cask phoenix
```

**Reply by jswhisperer** [reply]
brew install --force app 
worked for me without removing\
brew upgrade --force // also for all apps

**Reply by jswhisperer** [reply]
brew cleanup --repair after 
brew prune=all // maybe?

**Comment by ferenczy** [comment]
I'm getting the same error when trying to upgrade using the brew upgrade command. In my case, affected applications (casks) are double-commander, keepassxc, qgis and spark.

**Reply by paulochang** [reply]
@ferenczy this apps have their own update process. you can use the app update or the brew update but not both. If you want to use brew, uninstall them and re-install via brew.

**Reply by drogozan-r7** [reply]
But they have been installed using brew. Otherwise brew wouldn't be trying to update them when running `brew upgrade`. I install everything that's possible using brew. Or is it possible, that after those apps have been installed using brew, they have updated themselves and that has broken the brew upgrade process?

**Reply by p-linnane** [reply]
It shouldn't break the upgrade process, but by default apps with auto-updates aren't updated by `brew upgrade` unless you pass `--greedy`.

**Reply by 40detectives** [reply]
> @ferenczy this apps have their own update process. you can use the app update or the brew update but not both. If you want to use brew, uninstall them and re-install via brew.

That's simply not true. You can do as @p-linnane says for those kinds of apps and, for me, it had been working flawlessly until recently.

**Comment by loscrossos** [comment]
for anyone having this problem.. i had it with kate and with blender. using the --force flag didnt help.
I dont have a solution but a better problem description and a workaround:

the problem was that brew was trying to make a backup of the old version to `/opt/homebrew/Caskroom/` but there was a backup there.

Also it could not delete `/Applications/Blender.app`
so i did it manually (this on example of blender. i did the same for kate)

```
sudo rm -r -f /Applications/Blender.app  

sudo rm -r -f /opt/homebrew/Caskroom/blender/

brew remove blender  
```

then reinstalled blender (and kate) with no problems and the new version was picked.

**Comment by Phanquan** [comment]
i got this error when run `brew upgrade`, there were 4 apps that got the error, they functioned normally though but homebrew could not interact with them, then i had to uninstall them manually, and reinstall with homebrew

**Reply by jswhisperer** [reply]
did you try https://github.com/orgs/Homebrew/discussions/4758#discussioncomment-10757584

**Comment by mstgrv** [comment]
This issue seems to happen to me like this:

1. an app gets updated with `homebrew update`
2. it's not an app I use often, so I don't actually open that app yet
3. another update gets released but I still haven't opened it since step 1
4. run `homebrew update` again - new version detected but the update fails with the `It seems there is already an App at` error

**Reply by 40detectives** [reply]
> 2. it's not an app I use often, so I don't actually open that app yet
> 3. another update gets released but I still haven't opened it since step 1

This issue happened to me with the Google Chrome cask (among a few others). It's my default browser and one I use frequently, so that's probably not necessary to reproduce it

**Reply by jklap** [reply]
Second this related to Google Chrome -- it's started happening quite consistently -- can't create the backup because it already exists so the upgrade fails. I manually delete the backup and all is good... any then, usually within a week, Google Chrome will fail again. Note, I update brew DAILY
